src/overlays/actors/ovl_Bg_Crace_Movebg/z_bg_crace_movebg.c

```diff
@@ -28,47 +44,256 @@ const ActorInit Bg_Crace_Movebg_InitVars = {
+        }
+    }
+    return returnVal;
+}
 
-extern UNK_TYPE D_060003A0;
-extern UNK_TYPE D_06000E00;
+void func_80A70970(BgCraceMovebg* this, PlayState* play) {
+    this->unk164 = 0.0f;
+    this->unk160 = 0.0f;
+    this->actionFunc = func_80A7099C;
+    this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/BgCraceMovebg_Init.s")
+void func_80A7099C(BgCraceMovebg* this, PlayState* play) {
+    if (Flags_GetSwitch(play, BGCRACEMOVEBG_GET_7F0(&this->dyna.actor)) != 0) {
+        func_80A709E4(this, play);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/func_80A7090C.s")
+void func_80A709E4(BgCraceMovebg* this, PlayState* play) {
+    this->actionFunc = func_80A70A08;
+    this->unk164 = 180.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/func_80A70970.s")
+void func_80A70A08(BgCraceMovebg* this, PlayState* play) {
+    func_800B9010(&this->dyna.actor, NA_SE_EV_STONEDOOR_OPEN_S - SFX_FLAG);
+    Math_SmoothStepToF(&this->unk160, this->unk164, 2.0f, this->unk16C, 0.01f);
+    this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y + this->unk160;
+    if (this->unk160 == this->unk164) {
+        func_80A70A84(this, play);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/func_80A7099C.s")
+void func_80A70A84(BgCraceMovebg* this, PlayState* play) {
+    this->actionFunc = func_80A70A9C;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/func_80A709E4.s")
+void func_80A70A9C(BgCraceMovebg* this, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/func_80A70A08.s")
+void BgCraceMovebg_Destroy(Actor* thisx, PlayState* play) {
+    BgCraceMovebg* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/func_80A70A84.s")
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    if (!(this->unk170 & 0x8)) {
+        switch (this->dyna.actor.params & 0xF) {
+            case 0:
+            case 2:
+                Flags_UnsetSwitch(play, BGCRACEMOVEBG_GET_7F0(thisx));
+                Flags_UnsetSwitch(play, BGCRACEMOVEBG_GET_7F0(thisx) + 1);
+                break;
+            case 1:
+                Flags_UnsetSwitch(play, BGCRACEMOVEBG_GET_7F0(thisx));
+                break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/func_80A70A9C.s")
+void BgCraceMovebg_Update(Actor* thisx, PlayState* play) {
+    BgCraceMovebg* this = THIS;
+    s32 pad;
+    Player* player = GET_PLAYER(play);
+    s16 tempRot;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Crace_Movebg/BgCraceMovebg_Destroy.s")
+    switch (this->dyna.actor.params & 0xF) {
+        case 0:
+        case 2:
+            tempRot = this->dyna.actor.yawTowardsPlayer - this->dyna.actor.home.rot.y;
+            if ((tempRot >= -0x4000) && (tempRot < 0x4001)) {
```

Comment:
```suggestion
            if ((tempRot >= -0x4000) && (tempRot <= 0x4000)) {
```
Think this should OK